doc/directives.py

```diff
@@ -53,3 +56,166 @@ def run(self):
+                    x, y = self.options['size'].split(" ")
+
+                sphinx_gallery.gen_rst.scale_image(figname, save_figname,
+                                                   x, y)
+                # replace figure in rst with simple regex
+                thumbnail_rst = re.sub(r'..\sfigure::\s.*\.png',
+                                       '.. figure:: /{}'.format(save_figname),
+                                       thumbnail_rst)
+
+            thumbnail = StringList(thumbnail_rst.split('\n'))
+            thumb = nodes.paragraph()
+            self.state.nested_parse(thumbnail, self.content_offset, thumb)
+
+            return [thumb]
+        except FileNotFoundError as e:
+            print(e)
+            return []
+
+
+
+GALLERY_TEMPLATE = """
+.. raw:: html
+
+    <div class="sphx-glr-thumbcontainer" tooltip="{tooltip}">
+    
+.. only:: html
+
+    .. figure:: /{thumbnail}
+    
+        {description}
+        
+.. raw:: html
+
+    </div>
+"""
+
+
+class CustomGalleryItemDirective(Directive):
+    """Create a sphinx gallery style thumbnail.
+    tooltip and figure are self explanatory. Description could be a link to
+    a document like in below example.
+    Example usage:
+    .. customgalleryitem::
+        :tooltip: I am writing this tutorial to focus specifically on NLP for people who have never written code in any deep learning framework
+        :figure: /_static/img/thumbnails/babel.jpg
+        :description: :doc:`/beginner/deep_learning_nlp_tutorial`
+        :size: put image size here
+    If figure is specified, a thumbnail will be made out of it and stored in
+    _static/thumbs. Therefore, consider _static/thumbs as a 'built' directory.
+    """
+
+    required_arguments = 0
+    optional_arguments = 1
+    final_argument_whitespace = True
+    option_spec = {'tooltip': directives.unchanged,
+                   'figure': directives.unchanged,
+                   'description': directives.unchanged,
+                   'size': directives.unchanged}
+
+    has_content = False
+    add_index = False
+
+    def run(self):
+        try:
+            if 'tooltip' in self.options:
+                tooltip = self.options['tooltip'][:195]
+            else:
+                raise ValueError('tooltip not found')
+
+            if 'figure' in self.options:
+                env = self.state.document.settings.env
+                rel_figname, figname = env.relfn2path(self.options['figure'])
+                thumbnail = self.options['figure']
+                # thumbnail = os.path.join('_static/thumbs/', os.path.basename(figname))
+
+                # try:
+                #     os.makedirs('_static/thumbs')
+                # except FileExistsError:
+                #     pass
+
+                # x, y = (400, 280)
+                # if 'size' in self.options:
+                #     x, y = self.options['size'].split(" ")
+
+                # sphinx_gallery.gen_rst.scale_image(figname, thumbnail, int(x), int(y))
```

Comment:
Was this commented out from the walrus? If so, I can't remember why I did this. I think to avoid the thumbnails from automatically being scaled?

Reply:
We might as well delete it